wrappers/src/set_cs.cpp

```diff
@@ -261,4 +264,43 @@ REALM_EXPORT bool realm_set_set_equals(object_store::Set& set, const object_stor
     });
 }
 
+REALM_EXPORT Results* set_to_results(const object_store::Set& set, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        return new Results(set.as_results());
+    });
+}
+
+REALM_EXPORT Results* set_get_filtered_results(const object_store::Set& set, uint16_t* query_buf, size_t query_len, realm_value_t* arguments, size_t args_count, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        Utf16StringAccessor query_string(query_buf, query_len);
+        auto const& realm = set.get_realm();
+
+        auto query = set.get_query();
+
+        query_parser::KeyPathMapping mapping;
+        realm::populate_keypath_mapping(mapping, *realm);
+
+        std::vector<Mixed> mixed_args;
+        mixed_args.reserve(args_count);
+        for (size_t i = 0; i < args_count; ++i) {
+            if (arguments[i].type != realm_value_type::RLM_TYPE_LINK) {
+                mixed_args.push_back(from_capi(arguments[i]));
+            }
+            else {
+                mixed_args.push_back(from_capi(arguments[i].link.object, true));
+            }
+        }
+
+        Query parsed_query = set.get_table()->query(query_string, mixed_args, mapping);
+        DescriptorOrdering new_order;
+        if (auto parsed_ordering = parsed_query.get_ordering()) {
+            new_order.append(*parsed_ordering);
+        }
```

Comment:
I imagine List and Results have pretty much the same implementation as this one - can we extract it somewhere where it can be reused by all collections?

Reply:
Addressed with 17ce881 for now but I'd look into that again at some point to see if we can reduce the vast amount of arguments there (tomorrow or in another PR - whatever we prefer).